serialized_data_interface/local_sdi.py

```diff
@@ -19,42 +20,49 @@ def create_new_metadata(metadata):
     return metadata
 
 
-def change_zip_file(charm_name, metadata):
-    temp_dir = f"{charm_name}.tmp"
-    charm_zip = f"{charm_name}.charm"
+def change_zip_file(charm_name, metadata, charm_path="."):
+    try:
+        temp_dir = f"{charm_path}/{charm_name}.tmp"
+        charm_zip = f"{charm_path}/{charm_name}.charm"
 
-    os.mkdir(temp_dir)
+        os.mkdir(temp_dir)
```

Comment:
Would it be possible to use [tempfile](https://docs.python.org/3/library/tempfile.html) here? Specifically `tempfile.TemporaryDirectory`